### DISCUSSION TITLE
How do I remove auto comments on ENTER, o, and O?

### DISCUSSION THREAD

**Discussion by andradei** [LazyVim/LazyVim#431] [Q&A] [answered]
I tried following [this answer](https://neovim.discourse.group/t/disable-comment-auto-insertion/892/7) and added `vim.bo.formatoptions = jnql` to `config/options.lua` but it's not working. Where should I put it?

**Comment by dpetka2001** [comment, marked as answer]
I don't know if you specifically want to do it in `options.lua` but I have the following in `autocmds.lua` and works fine
```lua
augroup("mygroup", { clear = true })

autocmd("Filetype", {
  pattern = { "*" },
  callback = function()
    vim.opt.formatoptions = vim.opt.formatoptions - "o" -- Don't continue comments with o and O
  end,
  group = "mygroup",
  desc = "Don't continue comments with o and O",
})
```
I have it setup only for `o` and `O` disabling comments, so feel free to add for `Enter` as well.

**Reply by folke** [reply]
As I mentioned before, a lot of the default Neovim ftplugins overwrite any setting you do to formatoptions.

You can indeed use a FileType autocmd to override those

**Reply by andradei** [reply]
I pasted this code and got an error, but the stacktrace is longer than `nvim-notify` has space to show. How do I get the full error message so I can debug the code?

So far I had to append `vim.` to `augroup()` and `autocmd()`. But there are more errors.

**Reply by folke** [reply]
vim.api.nvim_create_autocmd
vim.api.nvim_create_augroup

Also, please just read NEovim docs. It's all in there.

**Reply by andradei** [reply]
Thanks. Being a beginner, most of the time I really don't know where to even look. So thanks for pointing me to `vim.api.nvim_create_*`

**Comment by dpetka2001** [comment]
Put this on the first lines
```lua
local augroup = vim.api.nvim_create_augroup
local autocmd = vim.api.nvim_create_autocmd
```
Forgot I changed it to my needs.

**Reply by andradei** [reply]
I figured as much, specially after @folke's reply. Thank you.

**Reply by dpetka2001** [reply]
Yes I was writing the message when he already replied and wasn't aware of that. There was no need for my answer.
